Userland/Libraries/LibGUI/AutocompleteProvider.cpp

```diff
@@ -168,7 +168,7 @@ void AutocompleteBox::apply_suggestion()
         return;
 
     auto selected_index = m_suggestion_view->selection().first();
-    if (!selected_index.is_valid())
+    if (!selected_index.is_valid() || selected_index.row() >= m_suggestion_view->model()->row_count())
```

Comment:
Is this happening because the language server sends an update to the client between the user selecting something from the autocomplete box and us trying to apply it? if so, maybe we should instead just invalidate the entire list and ignore the user's action?

Reply:
I see. Do you have an idea on how we can detect such a condition? (And sorry for the late reply..)

Reply:
the most simple would be to have a boolean that you set when the list is updated, and reset when you show the box; then checking it in `apply_suggestion()`.
if it's set, tell the user to go away :P

Reply:
yeah sound's good I'll do just that, thanks.
I think I'll also keep the index validation check, just in case 🤷‍♂️

Reply:
Thinking about it, I don't think that using that boolean would be appropriate, because `AutocompleteBox::update_suggestions` will normally be called while the autocomplete box is shown, for example if the user is typing with the autocomplete box up. 

I agree with you that the race is probably when we get an update from the LanguageServer between the user selecting something and `AutocompleteBox::apply_suggestion` being called, but i'm not sure how we can detect this :thinking: 

It seems like somehow we should have a way to invalidate a pending `apply_suggestions` event when we're in   `AutocompleteBox::update_suggestions`?

Reply:
:thinking: I don't think there's a way to cancel posted events right now
maybe your current way of "just don't crash, maybe being wrong is okay" is okay?

Reply:
Yeah, I think it's okay-ish for now (tm), and providing wrong autocomplete results is indeed not disastrous.
Anyways I added a FIXME about it in `AutocompleteBox::apply_suggestion()`.
I'm pulling this PR out of draft.